server/test/bookingTest.js

```diff
@@ -1,89 +1,192 @@
 process.env.NODE_ENV = 'test';
 require('dotenv').config({ path: './config/config.env' });
 const dbUtils = require('../utils/db');
-const moment = require('moment');
 
 const chai = require('chai');
 const server = require('../index');
 const should = chai.should();
 const chaiHttp = require("chai-http");
 const populateDb = require('../utils/populate')
+const moment = require('moment');
+let data;
 
 
 chai.use(chaiHttp);
 
-const book=async function () {
-    const newBooking = { lecture_id: 1, student_id: 2};
+const book = async function () {
+    let credentials = { email: data.students[0].email, password: data.students[0].password };
+    const agent = chai.request.agent(server);
+    let resLogin = await agent.post(`/api/login`).send(credentials);
+
+    let student_id = resLogin.body.id;
+
+    const newBooking = { lecture_id: 1, student_id: student_id };
     const tmp = `/api/students/${newBooking.student_id}/bookings`;
-    let res = await chai.request(server).post(tmp).send({lecture_id: newBooking.lecture_id});
+    let res = await agent.post(tmp).send({ lecture_id: newBooking.lecture_id });
     should.exist(res);
     res.body.should.be.an('object');
-    return res.status;
-        
+    return res;
+
 }
 
-const deletion =async function(){
-    const newBooking = { lecture_id: 1, student_id: 2};
-        const tmp = `/api/students/${newBooking.student_id}/lectures/${newBooking.lecture_id}`;
-        let res = await chai.request(server).delete(tmp).send();
-        should.exist(res);
-        res.body.should.be.an('object');
-        return res.status;
+const deletion = async function () {
+    let credentials = { email: data.students[0].email, password: data.students[0].password };
+    const agent = chai.request.agent(server);
+    let resLogin = await agent.post(`/api/login`).send(credentials);
+
+    let student_id = resLogin.body.id;
+    // console.log(student_id);
+
+    const newBooking = { lecture_id: 1, student_id: student_id };
+    const tmp = `/api/students/${newBooking.student_id}/lectures/${newBooking.lecture_id}`;
+    let res = await agent.delete(tmp).send();
+    should.exist(res);
+    res.body.should.be.an('object');
+    return res.status;
 
 }
 
 describe('Booking routes', function () {
-    before('create tables and clear db', async function() {
+    before('create tables and clear db', async function () {
         await dbUtils.reset();
     });
-
-    after('clear db', async function() {
-        await dbUtils.reset({ create: false });
+    before('populate db', async () => {
+        data = await dbUtils.populate();
     });
 
-    it('should allow student to book a lecture', async function() {
-        let res=await book();
-        res.should.be.equal(201);
+    it('should allow student to book a lecture', async function () {
```

Comment:
We should take a second look to this test:
- the login is duplicated (since it is already done in the book function)
- row 67 and 73 are the same, and they are not tested at all (the book function has its own parameters)